src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/RawlsGroupComponent.scala

```diff
@@ -137,7 +137,53 @@ trait RawlsGroupComponent {
         
         nextLevelUp.result.flatMap { nextGroups =>
           val nextCumulativeGroups = cumulativeGroups ++ nextGroups
-          listParentGroupsRecursive(nextGroups.toSet -- nextCumulativeGroups, nextCumulativeGroups)
+          listParentGroupsRecursive(nextGroups.toSet -- cumulativeGroups, nextCumulativeGroups)
+        }
+      }
+    }
+
+    private def listDescendantGroupsRecursive(groups: Set[RawlsGroupRecord], cumulativeGroups: Set[RawlsGroupRecord]): ReadAction[Set[RawlsGroupRecord]] = {
+      if (groups.isEmpty) DBIO.successful(cumulativeGroups)
+      else {
+        val nextLevelDown = for {
+          groupSubGroup <- groupSubgroupsQuery if (groupSubGroup.parentGroupName.inSetBind(groups.map(_.groupName)))
+          childGroup <- rawlsGroupQuery if (groupSubGroup.childGroupName === childGroup.groupName)
+        } yield childGroup
+
+        nextLevelDown.result.flatMap { nextGroups =>
+          val nextCumulativeGroups = cumulativeGroups ++ nextGroups
+          listDescendantGroupsRecursive(nextGroups.toSet -- cumulativeGroups, nextCumulativeGroups)
+        }
+      }
+    }
+
+    private def listGroupUsers(groupRec: RawlsGroupRecord): ReadAction[Set[RawlsUserRef]] = {
+      findUsersByGroupName(groupRec.groupName).result flatMap { userRecs =>
+        DBIO.successful(userRecs.map(u => RawlsUserRef(RawlsUserSubjectId(u.userSubjectId))).toSet)
+      }
+    }
+
+    def flattenGroupMembership(groupRef: RawlsGroupRef): ReadAction[Set[RawlsUserRef]] = {
+      load(groupRef) flatMap { groupOption =>
+        groupOption match {
```

Comment:
can remove `groupOption =>  groupOption match {` here.  Flatmaps can go directly to cases.